### ISSUE TITLE
No TestNG integration

### ISSUE THREAD

**Issue by novityFS** [cukespace/cukespace#50]
It seems that Cukespace does lack support for TestNG.

**Comment by rmannibucau** [comment]
Yes, do you want to hack it?

**Comment by novityFS** [comment]
Unfortunately I cannot. My knowledge is in using TestNG only, not internals.

**Comment by FibreFoX** [comment]
issue should stay open, maybe someone else will create support for TestNG (im lacking that knowledge too, but maybe it isnt that hard :smile: )

**Comment by rmannibucau** [comment]
All the logic is https://github.com/cukespace/cukespace/blob/master/core/src/main/java/cucumber/runtime/arquillian/ArquillianCucumber.java, just need to be adapted to Arquillian TestNG and TestNG internals

**Comment by greggvarona** [comment]
I am working on a rough implementation as proof of concept that Cukespace can be used together with TestNG. I still have problems with the reporting though. It says that there were no features in the .json report file, bec. the json file is empty ;( . But, if anyone wants to have a look and give feedback...

Link to the branch: [https://github.com/greggvarona/cukespace/tree/testng](https://github.com/greggvarona/cukespace/tree/testng)

**Comment by rmannibucau** [comment]
Pushed the work I did on testng some time ago, there is a examples-testng module to try to share the overall idea. I didn't manage to find time to test it as much as required (typically the persistence module is not tested at all, even manually) but can be a starting point for this feature.

Hope it helps. Feel free to say it is useless and i could revert it, no problem ;)

**Comment by celarcosa** [comment]
I tried running the recent changes, but it produced a problem in the reports. The JSON data gets written twice producing a MalformedJsonException. I was able to resolve this by removing the printing done in CukeSpace TestNG's runFeature(), a process that's also performed by its caller method, BaseCukeSpace.runFeatures().

For reference, I made a pull request in https://github.com/cukespace/cukespace/pull/79.
